neqo-common/src/datagram.rs

```diff
@@ -90,7 +94,6 @@ fn fmt_datagram() {
     let d = datagram([0; 1].to_vec());
     assert_eq!(
         format!("{d:?}"),
-        "Datagram IpTos(Cs0, NotEct) TTL Some(128) [fe80::1]:443->[fe80::1]:443: [1]: 00"
-            .to_string()
+        "Datagram IpTos(Cs0, Ect0) TTL Some(128) [fe80::1]:443->[fe80::1]:443: [1]: 00".to_string()
```

Comment:
Rather than call to_string(), you should compare &format!(...) with the &str.  It saves an allocation.  (In tests, no big deal, but it's a good habit.)